### DISCUSSION TITLE
Can bounce mails also be sent to the defined routes?

### DISCUSSION THREAD

**Discussion by Pacerino** [postalserver/postal#2787] [Help with using Postal]
We like the functions of Postal, but what we are missing is the option of an IMAP/POP3 mailbox. Many applications such as Listmonk, Mautic etc. use IMAP to process bounces. I have managed to send all incoming mails via SMTP route to a Postfix and then retrieve them using Dovecot. However, bounces that are mapped to an original message by Postal are not sent to the SMTP route. Is it possible to configure this so that Postal also sends the bounce mails to the routes?

Of course I could also adapt the respective application to Postal's bounce handler using the HTTP API and possibly write a plugin for it, but that would exceed my workload.

Edit:
For my understanding from this line https://github.com/postalserver/postal/blob/main/app/jobs/unqueue_message_job.rb#L100 if the message could be linked to a original message as a bounce, the whole '_send it to a route function_' from https://github.com/postalserver/postal/blob/main/app/jobs/unqueue_message_job.rb#L222 will be ignored. Is my assumption correct?

**Comment by willpower232** [comment]
Unfortunately Postal isn't really intended to keep messages long term so wrangling a traditional mailbox would imply extra storage concerns which are not part of Postal currently.

Bounces need a route named `__returnpath__` currently as per https://docs.postalserver.io/other/auto-responders-and-bounces

I'm sure there is a good reason bounce messages apparently don't work on SMTP routes but I don't know what it is.
